parlai/core/dict.py

```diff
@@ -294,8 +306,10 @@ def txt2vec(self, text, vec_type=np.ndarray):
                 (self[token] for token in self.tokenize(str(text))),
                 np.int
             )
-        else:
+        elif vec_type == list:
```

Comment:
need to make sure you are not breaking anything else that uses the dict, e.g. IR baseline etc.